Mathlib/Algebra/Polynomial/FieldDivision.lean

```diff
@@ -658,6 +659,28 @@ theorem irreducible_iff_lt_natDegree_lt {p : R[X]} (hp0 : p ≠ 0) (hpu : ¬ IsU
   simp only [IsUnit.dvd_mul_right
     (isUnit_C.mpr (IsUnit.mk0 (leadingCoeff p)⁻¹ (inv_ne_zero (leadingCoeff_ne_zero.mpr hp0))))]
 
+open UniqueFactorizationMonoid in
+/--
+Every polynomial over a field factors to some constant times the product of some
+irreducible monic polynomials.
+-/
+theorem exists_multiset_irreducible_and_monic_and_mul_prod_eq (a : R[X]) :
+    ∃ (f : Multiset R[X]) (v : R), (∀ b ∈ f, Irreducible b ∧ b.Monic) ∧ C v * f.prod = a := by
+  classical
+  by_cases ha : a = 0
```

Comment:
`normalizedFactors 0` is also empty, right? Maybe you can just use that instead of the unspecified f?